torch/csrc/distributed/c10d/symm_mem/nvshmem_extension.cuh

```diff
@@ -3,7 +3,15 @@
 #include <c10/macros/Macros.h>
 #include <ATen/ATen.h>
 
-#define NVSHMEM_CHECK(stmt, msg)                                             \
+#define NVSHMEM_CHECK(stmt)                                             \
+  do {                                                                  \
+    int result = (stmt);                                                \
+    TORCH_CHECK(                                                        \
+        result == 0,                                                    \
+        "NVSHMEM call failed, error code: " + std::to_string(result));  \
```

Comment:
Use lazy string construction and add the to_string as the third argument to the macro instead of concatenation it eagerly.